src/Servers/Kestrel/Core/src/Internal/Http2/Http2HeadersEnumerator.cs

```diff
@@ -90,11 +90,29 @@ public bool MoveNext()
         else
         {
             return _genericEnumerator!.MoveNext()
-                ? SetCurrent(_genericEnumerator.Current.Key, _genericEnumerator.Current.Value, default)
+                ? SetCurrent(_genericEnumerator.Current.Key, _genericEnumerator.Current.Value, GetKnownRequestHeaderType(_genericEnumerator.Current.Key))
                 : false;
         }
     }
 
+    private static KnownHeaderType GetKnownRequestHeaderType(string headerName)
+    {
+#if DEBUG
```

Comment:
This is only for debugging, it doesn't affect the tests?

Reply:
It does, but in a way that tests work with and without it.

How do we feel about having code in product that is only used by tests?

Reply:
Can we just copy this version of `Http2HeadersEnumerator` (and probably `HPackHeaderWriter`) into the test project and use those from `Http2TestBase` instead of the real versions?